File: api-form/Modelo/Producto.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace api_form.Modelo
{
    [Table("Productos")]
    public class Producto
    {
        [Key]
        [Column("ID")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }

        [Column("Nombre")]
        public string Nombre { get; set; }

        [Column("Unidades")]
        public string Unidades { get; set; }

        [Column("Artista")]
        public string Artista { get; set; }

        [Column("Imagen")]
        public string Imagen { get; set; }

        [Column("Precio")]
        public string Precio { get; set; }

        [Column("Genero")]
        public string Genero { get; set; }

        [Column("Descripcion")]
        public string Descripcion { get; set; }

        public Producto() { }
    }
}

File: aplicacion-web-sergio-react-vite/src/api/producto.ts
import api from './client';

// Simplified interface based on the provided API shape
export interface Producto {
  id: number;
  nombre: string;
  unidades: string;
  artista: string;
  imagen: string;
  precio: string;
  genero: string;
  descripcion: string;
}

// Keep payload as-is to match current backend expectations when creating/updating
export interface ProductoPayload {
  Nombre: string;
  Artista: string;
  Genero: string;
  Descripcion: string;
  Imagen: string;
  Precio: string; // Backend expects string per current implementation
  Unidades: string; // Backend expects string
  ID?: number;
}

export const ProductoApi = {
  async list(): Promise<Producto[]> {
    const res = await api.get('/Producto');
    return res.data ?? [];
  },

  async get(id: string | number): Promise<Producto> {
    const res = await api.get(`/Producto/${id}`);
    return res.data;
  },

  async create(payload: ProductoPayload): Promise<any> {
    const res = await api.post('/Producto', payload);
    return res.data;
  },

  async update(id: string | number, payload: ProductoPayload): Promise<any> {
    const res = await api.put(`/Producto/${id}`, payload);
    return res.data;
  },

  async remove(id: string | number): Promise<any> {
    const res = await api.delete(`/Producto/${id}`);
    return res.data;
  },
};
